2.Portfolio/Portofilo/ViewController.swift

```diff
@@ -18,36 +26,25 @@ class ViewController: UIViewController,UIScrollViewDelegate {
     @IBOutlet var btnimg2: UIImageView!
     @IBOutlet var btnimg3: UIImageView!
     
-    
-    
-
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view, typically from a nib.
-        scrollview.contentSize = CGSizeMake(320, 1400)
-        //将scrollview委托给其他
-        scrollview.delegate = self
-        scrollview.showsVerticalScrollIndicator = false
-        
-        
-        
     }
     
     func scrollViewDidScroll(scrollView: UIScrollView) {
         println(scrollview.contentOffset.y)
-        var scrolly = scrollview.contentOffset.y
-        if(scrolly < 0){
-        topview.transform = CGAffineTransformMakeScale(1-scrolly/100,1-scrolly/100)
-        topview.frame.origin.y = scrolly/2
+        let scrolly = scrollview.contentOffset.y
+        
+        if scrolly < 0 {
+            topview.transform = CGAffineTransformMakeScale(1 - scrolly / 100, 1 - scrolly / 100)
+            topview.frame.origin.y = scrolly / 2
         }
-        else if(scrolly > 0){
-            blurview.alpha = 0+scrolly/150
-            topview.frame.origin.y = -scrolly/5
+        else if scrolly > 0 {
+            blurview.alpha = min(1, scrolly / 150)
+            topview.frame.origin.y = max(-topview.frame.height / 5, -scrolly / 5)
```

Comment:
If self.view has very long height, scrolly has very high value.

It is better to define value limit by using "max" function.